### PR TITLE
[Feature] Asset Provider

### PR DESCRIPTION
**Pull request by hopeman15**
name: Asset Provider
about: A crazy first approach for accessible assets 😅 
label: 'WIP'

---

## Description
This is just an idea to handle assets in a more accessible way. There are a few things that I still don't like, such as passing the android context around. I did limit this as best as possible and would work to eliminate it altogether so that one can access the assets without it. `ChaiImage.DCKE_MAIN` vs. `ChaiImage.DCKE_MAIN(ctx)`.

⚠️ I didn't want to spend too much more time on this if this isn't a viable solution or something we want to pursue further.

Anyway, let me know what you think 🙌 Let's discuss this one 🤓 

---
## How can this PR be tested?
You can test it in a few ways:
* Run the tests 🧪 
* Run ChaiDemo, here you can see the idea in action 🚀 

---

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Ensure your branch is up to date with develop branch.
- [ ] Ensure your branch runs without breaking.
- [ ] Ensure you solved any merge conflicts.
- [ ] Smaller PRs for the ease of reviewing
- [ ] Make sure to open a GitHub issue as a bug/feature request before writing your code! This means that we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass (`./gradlew --init-script gradle/init.gradle.kts spotlessApply` to automatically apply formatting)
- [ ] Appropriate docs were updated (You are adding code so...)
- [ ] Tests have been written

Is this your first Pull Request?
- [ ] Run `./tools/setup.sh`
- [ ] Import the code formatting style as explained in [the setup script](/tools/setup.sh#L40).

Fixes #<issue_number_goes_here> 🦕

# Screenshot
Screenshot_20230324_204158

### CODE REVIEW

File: chai/src/main/kotlin/com/droidconke/chai/assets/AssetProvider.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.drawable.BitmapDrawable
+import androidx.core.graphics.drawable.toBitmap
+import com.droidconke.chai.exceptions.attemptAssetRetrieval
+import java.io.InputStream
+
+object AssetProvider {
+
+    fun image(ctx: Context, path: String): Bitmap =
```
Comment (hopeman15) [inline chai/src/main/kotlin/com/droidconke/chai/assets/AssetProvider.kt:12]: I want to add support for other asset types, using generics to reduce the amount of redundancy in creating a bunch of new functions

**Comment by tamzi** [comment]
@hopeman15 are we merging this or leaving a WIP tag on it for a future reWork?

**Comment by hopeman15** [comment]
> @hopeman15 are we merging this or leaving a WIP tag on it for a future reWork?

So this isn't done 🙈 I wanted your feedback first to decide whether or not it makes sense to pursue this. Is this what you were thinking, or doe this go in a completely different direction?

**Comment by tamzi** [comment]
LGTM 👋🏿

**Comment by tamzi** [comment]
or in full..makes sense to pursue it! 😄

**Pull request closed without merging**